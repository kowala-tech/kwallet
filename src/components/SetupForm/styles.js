const styles = (theme) => ({
	root: {
		width: "100%",
		maxWidth: "500px"
	},
	button: {
		marginTop: theme.spacing.unit,
		marginRight: theme.spacing.unit,
	},
	stepper: {
		paddingLeft: 0,
		paddingRight: 0
	},
	actionsContainer: {
		textAlign: "right",
		marginBottom: theme.spacing.unit * 2,
	},
	pendingContainer: {
		display: "flex",
		alignItems: "center",
	},
	progress: {
		marginRight: theme.spacing.unit * 2,
		minWidth: 24
	},
});

export default styles;
