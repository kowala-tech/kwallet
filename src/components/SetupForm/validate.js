const validate = values => {
	const errors = {};
	// Username
	if (!values.username) {
		errors.username = "Username is required.";
	} else if (values.username.length < 8) {
		errors.username = "Username must be at least 8 characters.";
	}
	// Password
	if (!values.password) {
		errors.password = "Password is required.";
	} else if (values.password.length < 8) {
		errors.password = "Password must be at least 8 characters.";
	}
	// Password Confirmation
	if (values.password !== values.passwordConfirmation) {
		errors.passwordConfirmation = "Passwords do not match.";
	}
	// Pin
	if (!values.pin) {
		errors.pin = "PIN is required.";
	} else if (values.pin.length !== 4) {
		errors.pin = "PIN must 4 digits.";
	}
	// Pin Confirmation
	if (values.pin !== values.pinConfirmation) {
		errors.pinConfirmation = "PINs do not match.";
	}
	return errors;
};

export default validate;
