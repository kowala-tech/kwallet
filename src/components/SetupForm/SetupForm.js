import React from "react";
import PropTypes from "prop-types";
import { compose } from "recompose";
// Material UI Imports
import { withStyles } from "@material-ui/core/styles";
import Grow from "@material-ui/core/Grow";
import Fade from "@material-ui/core/Fade";
import Stepper from "@material-ui/core/Stepper";
import Step from "@material-ui/core/Step";
import StepLabel from "@material-ui/core/StepLabel";
import StepContent from "@material-ui/core/StepContent";
import Typography from "@material-ui/core/Typography";
import Card from "@material-ui/core/Card";
import CardContent from "@material-ui/core/CardContent";
import CircularProgress from "@material-ui/core/CircularProgress";
// Component Related Imports
import styles from "./styles";
import UsernameForm from "./components/UsernameForm";
import PasswordForm from "./components/PasswordForm";
import PinForm from "./components/PinForm";

function getStepContent(step, next, back, sumbit) {
	switch (step) {
	case 0:
		return <UsernameForm onSubmit={next}/>;
	case 1:
		return (<PasswordForm
			onSubmit={next}
			previousStep={back}
		/>);
	case 2:
		return (<PinForm
			onSubmit={sumbit}
			previousStep={back}
		/>);
	default:
		return "Unknown step";
	}
}

class SetupForm extends React.Component {
	constructor(props) {
		super(props);
		this.state = {
			activeStep: 0,
		};
	}

	handleNext = () => {
		this.setState({ activeStep: this.state.activeStep + 1 });
	};

	handleBack = () => {
		this.setState({ activeStep: this.state.activeStep - 1 });
	};

	handleSubmit = (data) => {
		this.handleNext();
		this.props.onSubmit(data);
	}

	render() {
		const { classes } = this.props;
		const { activeStep } = this.state;

		const steps = [
			"Select your username",
			"Create your password",
			"Create your 4-digit PIN"
		];

		return (
			<Grow in>
				<Card className={classes.root}>
					<CardContent>
						<Typography variant="subheading">Create A New Account</Typography>
						<Stepper className={classes.stepper}
							activeStep={activeStep}
							orientation="vertical"
							elevation={0}>
							{steps.map((label, index) => {
								return (
									<Step key={label}>
										<StepLabel>{label}</StepLabel>
										<StepContent>
											{getStepContent(
												index,
												this.handleNext,
												this.handleBack,
												this.handleSubmit
											)}
										</StepContent>
									</Step>
								);
							})}
						</Stepper>
						{activeStep === steps.length && (
							<Fade in>
								<div className={classes.pendingContainer}>
									<CircularProgress className={classes.progress}/>
									<Typography variant="caption"
										className={classes.pendingText}>Your account is being created. Depending on the speed of your device, this process may take a few moments.</Typography>
								</div>
							</Fade>
						)}
					</CardContent>
				</Card>
			</Grow>
		);
	}

}

SetupForm.propTypes = {
	classes: PropTypes.object.isRequired,
	onSubmit: PropTypes.func.isRequired
};

export default compose(
	withStyles(styles)
)(SetupForm);
