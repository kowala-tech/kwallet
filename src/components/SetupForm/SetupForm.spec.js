import React from "react";
import { shallow } from "enzyme";
import toJson from "enzyme-to-json";
// Story related component imports
import SetupForm from "./SetupForm";

describe("SetupForm", () => {

	it("it should match the snapshot", () => {
		const component = toJson(
			shallow(
				<SetupForm
					onSubmit={jest.fn()}
				/>
			)
		);
		expect(component).toMatchSnapshot();
	});

});
