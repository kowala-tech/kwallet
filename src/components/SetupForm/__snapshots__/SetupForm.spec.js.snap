// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SetupForm it should match the snapshot 1`] = `
<SetupForm
  classes={
    Object {
      "actionsContainer": "SetupForm-actionsContainer-4",
      "button": "SetupForm-button-2",
      "pendingContainer": "SetupForm-pendingContainer-5",
      "progress": "SetupForm-progress-6",
      "root": "SetupForm-root-1",
      "stepper": "SetupForm-stepper-3",
    }
  }
  onSubmit={[MockFunction]}
/>
`;
